test/middleware/express.test.ts

```diff
@@ -183,154 +191,216 @@ xdescribe('Express Middleware tests', () => {
-                expect(mockResponse.locals?.complexity).toBeGreaterThanOrEqual(0);
+                expect(mockResponse.locals?.graphqlGate).toEqual(expected);
+                expect(mockResponse.locals?.graphqlGate.complexity).toBeGreaterThanOrEqual(0);
 
-                expect(mockResponse.locals).toHaveProperty('timestamp');
-                expect(mockResponse.locals?.timestamp).toBeInstanceOf('number');
                 // confirm that this is timestamp +/- 5 minutes of now.
                 const now: number = Date.now().valueOf();
-                const diff: number = Math.abs(now - (mockResponse.locals?.timestamp || 0));
-                expect(diff).toBeLessThan(5 * 60);
+                const diff: number = Math.abs(
+                    now - (mockResponse.locals?.graphqlGate.timestamp || 0)
+                );
+                expect(diff).toBeLessThan(5 * 60 * 1000);
             });
         });
 
         describe('Correctly limits requests', () => {
             describe('Allows requests', () => {
-                test('...a single request', () => {
+                test('...a single request', async () => {
                     // successful request calls next without any arguments.
-                    middleware(mockRequest as Request, mockResponse as Response, nextFunction);
+                    await middleware(
+                        mockRequest as Request,
+                        mockResponse as Response,
+                        nextFunction
+                    );
                     expect(nextFunction).toBeCalledTimes(1);
                     expect(nextFunction).toBeCalledWith();
                 });
 
-                test('Multiple valid requests at > 1 second intervals', () => {
+                test('Multiple valid requests at > 10 second intervals', async () => {
+                    const requests = [];
                     for (let i = 0; i < 3; i++) {
-                        const next: NextFunction = jest.fn();
-                        middleware(complexRequest as Request, mockResponse as Response, next);
-                        expect(next).toBeCalledTimes(1);
-                        expect(next).toBeCalledWith();
-
-                        // advance the timers by 1 second for the next request
-                        jest.advanceTimersByTime(1000);
+                        requests.push(
+                            middleware(
+                                complexRequest as Request,
+                                mockResponse as Response,
+                                nextFunction
+                            )
+                        );
+                        // advance the timers by 10 seconds for the next request
+                        jest.advanceTimersByTime(10000);
+                    }
+                    await Promise.all(requests);
+                    expect(nextFunction).toBeCalledTimes(3);
+                    for (let i = 1; i <= 3; i++) {
```

Comment:
Is this for loop with `nthCalledWith` necessary? Line 291 checks that it was called 3 times, does this check if it was called with no arguments?

Reply:
It's not completely necessary but a good check to have. line 291 `toBeCalledTimes` only checks how many times the function was called. The loop with `nthCalledWith` checks that no arguments were provided to next. I wasn't able to come up with a cleaner way to do this after refactoring  to not use `await` in the for loop